src/sparseset.jl

```diff
@@ -0,0 +1,86 @@
+
+const EMPTY_VEC = Memory{Int}(undef, 0)
+
+mutable struct SparseSetStructVector{C} <: AbstractIndexedStructVector
+    idvec::Memory{Int}
+    last_id::Int
+    const components::C
+end
+
+function SparseSetStructVector(components::NamedTuple)
+    allequal(length.(values(components))) || error("All components must have equal length")
+    len = length(first(components))
+    comps = merge((ID=collect(1:len),), components)
+    SparseSetStructVector{typeof(comps)}(EMPTY_VEC, len, comps)
+end
+
+@inline function id_to_index(isv::SparseSetStructVector, id::Int)
+    idvec = getfield(isv, :idvec)
+    idvec_len = length(idvec)
+    idvec_len == 0 && return 1 <= id <= lastkey(isv) ? id : throw(KeyError(id))
+    idvec_len < id && throw(KeyError(id))
```

Comment:
```suggestion
    1 <= id <= idvec_len || throw(KeyError(id))
```

This was missing the zero or negative id case

Reply:
good catch, thank you